ledger/acctupdates.go

```diff
@@ -386,8 +405,16 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 		// on-disk DB.
 
 		persistedData, err := au.accountsq.lookupKeyValue(key)
+		if err != nil {
+			return nil, err
+		}
+
 		if persistedData.round == currentDbRound {
-			return persistedData.value, nil
+			// if we read actual data return it. This includes deleted values
+			// where persistedData.value == nil to avoid unnecessary db lookups
+			// for deleted KVs.
+			au.baseKVs.writePending(persistedData, key)
+			return persistedData.value, err
```

Comment:
Now that `err` is checked earlier, let's make this explicitly `nil` so it is obvious it's the good path.

Reply:
got it, fixed.